modules/API/pages/gsql-endpoints.adoc

```diff
@@ -0,0 +1,4415 @@
+
+`GET /gsql/v1/schema/vertices/{vertexName}`
+
+This endpoint is used to show a local vertex.
+
+==== Example
+
+[,tabs]
+====
+Request::
++
+--
+[source,bash]
+----
+curl -X GET "http://localhost:14240/gsql/v1/schema/vertices/house?graph=g"
+----
+--
+Response::
++
+--
+[source.wrap,console]
+----
+{"error":false,"message":"","results":{"Config":{"STATS":"OUTDEGREE_BY_EDGETYPE"},"IsLocal":true,"Attributes":[{"AttributeType":{"Name":"STRING"},"AttributeName":"name"}],"PrimaryId":{"AttributeType":{"Name":"UINT"},"AttributeName":"id"},"Name":"house"}}
+----
+--
+====
+
+==== Parameters:
+//[width="100%",cols="25%,25%,50%",options="header",]
+[%autowidth]
+|===
+| Name | Required | Description
+
+| `graph`
+| yes
+| the vertices on which graph.
+|===
+
+'''        
+=== show a global vertex
+
+`GET /gsql/v1/schema/vertices/{vertexName}`
+
+This endpoint is used to show a global vertex
+
+==== Example
+
+[,tabs]
+====
+Request::
++
+--
+[source,bash]
+----
+curl -X GET "http://localhost:14240/gsql/v1/schema/vertices/person"
+----
+--
+Response::
++
+--
+[source.wrap,console]
+----
+{"error":false,"message":"","results":{"Config":{"STATS":"OUTDEGREE_BY_EDGETYPE"},"Attributes":[{"AttributeType":{"Name":"UINT"},"AttributeName":"sid"},{"AttributeType":{"Name":"STRING"},"AttributeName":"name"}],"PrimaryId":{"AttributeType":{"Name":"UINT"},"AttributeName":"id"},"Name":"person"}}
+----
+--
+====
+
+==== Parameters:
+
+NONE
+
+'''     
+
+=== create vertex usgin gsql command 
+
+`POST /gsql/v1/schema/vertices`
+
+This endpoint is used to create global vertices using gsql command
+
+==== Example
+
+[,tabs]
+====
+Request::
++
+--
+[source,bash]
+----
+curl -X POST "http://localhost:14240/gsql/v1/schema/vertices?gsql=true" -d '{"gsql":["CREATE VERTEX UserA (PRIMARY_ID user_id UINT, name STRING)", "CREATE VERTEX UserB (PRIMARY_ID user_id UINT, name STRING)"]}'
+----
+--
+Response::
++
+--
+[source.wrap,console]
+----
+{"error":false,"message":"Successfully create vertices: [UserA, UserB]"}
+----
+--
+====
+==== Parameters:
+
+//[width="100%",cols="25%,25%,50%",options="header",]
+[%autowidth]
+|===
+| Name | Required | Description
+
+| `gsql`
+| yes
+| this should be true, means gsql command foramt
+|===
+
+
+'''     
+
+=== create vertex (add global vertex to local graph)
+
+`POST /gsql/v1/schema/vertices`
+
+This endpoint is used to add global vertices to local graph
+
+==== Example
+
+[,tabs]
+====
+Request::
++
+--
+[source,bash]
+----
+curl -X POST -H "content-type: application/json" "http://localhost:14240/gsql/v1/schema/vertices?gsql=false&graph=empty_graph" -d '{"addVertices":["user","website"]}'
+----
+--
+Response::
++
+--
+[source.wrap,console]
+----
+{"error":false,"message":"Successfully add vertices: [user, website] on graph empty_graph"}
+----
+--
+====
+
+==== Parameters:
+
+//[width="100%",cols="25%,25%,50%",options="header",]
+[%autowidth]
+|===
+| Name | Required | Description
+
+| `gsql`
+| yes
+| this should be false, means a json format

```

Comment:
Not clear, should be:

`False means payload would be json format, True means payload would be query string, which is the gsql command to add global vertex to local graph.`